engine/apps/user_management/tests/test_sync.py

```diff
@@ -399,38 +416,39 @@ def test_sync_organization_is_rbac_permissions_enabled_cloud(
         },
     )
 
+    mock_gcom_client.return_value.is_stack_active.return_value = gcom_api_response
+
+    # TODO: we should refactor this so that's there not as much duplication on this patching across tests...
+    mock_grafana_api_client.return_value.get_users.return_value = api_users_response
+    mock_grafana_api_client.return_value.get_teams.return_value = (api_teams_response, None)
+    mock_grafana_api_client.return_value.get_team_members.return_value = (api_members_response, None)
+    mock_grafana_api_client.return_value.get_grafana_incident_plugin_settings.return_value = (
+        {"enabled": True, "jsonData": {"backendUrl": MOCK_GRAFANA_INCIDENT_BACKEND_URL}},
+        None,
+    )
+    mock_grafana_api_client.return_value.get_grafana_labels_plugin_settings.return_value = (
+        {"enabled": True, "jsonData": {}},
+        None,
+    )
+    mock_grafana_api_client.return_value.check_token.return_value = (None, {"connected": True})
+    mock_grafana_api_client.return_value.is_rbac_enabled_for_organization.return_value = grafana_api_response
+
     random_uuid = "random"
     with patch("apps.user_management.sync.uuid.uuid4", return_value=random_uuid):
-        with patch.object(GrafanaAPIClient, "check_token", return_value=(None, api_check_token_call_status)):
-            with patch.object(GrafanaAPIClient, "get_users", return_value=api_users_response):
-                with patch.object(GrafanaAPIClient, "get_teams", return_value=(api_teams_response, None)):
-                    with patch.object(GrafanaAPIClient, "get_team_members", return_value=(api_members_response, None)):
-                        with patch.object(
-                            GrafanaAPIClient,
-                            "get_grafana_incident_plugin_settings",
-                            return_value=(
-                                {"enabled": True, "jsonData": {"backendUrl": MOCK_GRAFANA_INCIDENT_BACKEND_URL}},
-                                None,
-                            ),
-                        ):
-                            with patch.object(
-                                GrafanaAPIClient,
-                                "get_grafana_labels_plugin_settings",
-                                return_value=(
-                                    {"enabled": True, "jsonData": {}},
-                                    None,
-                                ),
-                            ):
```

Comment:
I think I'll also refactor the tests in this file to use a fixture, rather than 9 nested context managers 😆